### ISSUE TITLE
IllegalArgumentException when parsing null value from DB into  Option[scala.math.BigDecimal]

### ISSUE THREAD

**Issue by williamhaw** [AugustNagro/magnum#20]
# Issue

For `Option[scala.math.BigDecimal]`, in [line 301 of DbCodec](https://github.com/AugustNagro/magnum/blob/master/magnum/src/main/scala/com/augustnagro/magnum/DbCodec.scala#L301), it delegates to ScalaBigDecimalCodec.readSingle.
At [Line 281 of DbCodec](https://github.com/AugustNagro/magnum/blob/master/magnum/src/main/scala/com/augustnagro/magnum/DbCodec.scala#L281), `rs.getBigDecimal(pos)` returns null, which when passed to the constructor of scala.math.BigDecimal causes it to throw IllegalArgumentException.

# To reproduce:

Setup of table and data:
```sql
CREATE TABLE IF NOT EXISTS test(
    test_id INT NOT NULL,
    my_big_decimal NUMERIC,
    PRIMARY KEY (test_id)
);

INSERT INTO test(test_id) VALUES (1);
```

Scala representation:
```scala
case class Test(@Id() testId: Int, myBigDecimal: Option[BigDecimal])
```

Query:
```scala
connect(ds)(sql"SELECT * FROM test WHERE test_id = 1".query[Test].run())
```

Result:
```
IllegalArgumentException("null value for BigDecimal")
```

# Workaround
Declare field as Option[java.math.BigDecimal] instead and convert manually later.

**Comment by williamhaw** [comment]
Would changing OptionCodec.readSingle() to perform rs.getObject, then calling codec.readSingle() only if the result is not null be efficient enough?

**Comment by AugustNagro** [comment]
Thanks for the report, I'll take a look soon!

**Comment by AugustNagro** [comment]
> If OptionCodec.readSingle() is changed to perform rs.getObject(pos), then call codec.readSingle() only if the result is not null, would this be efficient enough?

This is a good idea; but since we already propagate null in the case of the other type classes, it should be enough to do the same for ScalaBigDecimalCodec.

Please review the above MR, and then I will release v1.1.1

**Comment by williamhaw** [comment]
LGTM! Thanks!

**Comment by AugustNagro** [comment]
Released v1.1.1: https://github.com/AugustNagro/magnum/releases/tag/v1.1.1
